server/app/controllers/FileController.java

```diff
@@ -64,13 +71,38 @@ public CompletionStage<Result> show(Request request, long applicantId, String fi
             });
   }
 
+  private long extractApplicantIdFromFileKey(String fileKey) {
+    String applicantString = fileKey.split("/")[0];
+    long applicantid = (long) Long.parseLong(applicantString.split("-")[1].trim());
+    return applicantid;
+  }
+
   @Secure(authorizers = Authorizers.Labels.ANY_ADMIN)
   public Result adminShow(Request request, long programId, String fileKey) {
     try {
+      boolean isAuthorized = false;
+      if (fileKey != null) {
+        // if the applicant has submitted the same file to both programA and program-B
+        // then the admin has authorization to access the file in programA
+        long applicationId = extractApplicantIdFromFileKey(fileKey);
+        ImmutableList<Application> applicationList =
+            this.applicationRepository.getAllApplications().stream()
+                .filter(application -> application.id == applicationId)
+                .collect(toImmutableList());
```

Comment:
issue (blocking): the file key has the applicant ID embedded in it, not the application ID https://github.com/seattle-uat/civiform/blob/main/server/app/services/cloud/FileNameFormatter.java#L35

Reply:
a possible longer-term fix for this could be to add a column to the files table that stores an array of program IDs that can access the file. when the applicant submits an application the server can scan the application for file upload questions and ensure the program has access.